core/src/main/kotlin/io/specmatic/core/pattern/StringPattern.kt

```diff
@@ -67,36 +67,27 @@ data class StringPattern (
         return JSONArrayValue(valueList)
     }
 
-    private val randomStringLength: Int =
+    private val patternMinLength: Int =
         when {
-            minLength != null && 5 < minLength -> minLength
-            maxLength != null && 5 > maxLength -> maxLength
+            minLength != null && minLength > 0 -> minLength
+            maxLength != null && maxLength < 5 -> 1
             else -> 5
         }
-    
-    override fun generate(resolver: Resolver): Value {
-        val defaultExample: Value? = resolver.resolveExample(example, this)
-
-        if (regex != null) {
-            if(defaultExample == null)
-                return StringValue(Generex(regex.removePrefix("^").removeSuffix("$")).random(randomStringLength))
-
-            val defaultExampleMatchResult = matches(defaultExample, resolver)
 
-            if(defaultExampleMatchResult.isSuccess())
-                return defaultExample
-
-            throw ContractException("Schema example ${defaultExample.toStringLiteral()} does not match pattern $regex")
-        }
-
-        if(defaultExample != null) {
-            if(defaultExample !is StringValue)
-                throw ContractException("Schema example ${defaultExample.toStringLiteral()} is not a string")
+    override fun generate(resolver: Resolver): Value {
+        val defaultExample = resolver.resolveExample(example, this)
 
-            return defaultExample
+        defaultExample?.let {
+            if (matches(it, resolver).isSuccess()) {
+                return it
+            }
+            throw ContractException("Schema example ${it.toStringLiteral()} does not match pattern $regex")
```

Comment:
In this form, this condition is no more about regex matching.

Ideally, the resolver should be given a new MismatchMessages indicating that we are checking the schema example.

The `matches` method returns a result, and we could call result.throwOnFailure() instead of throwing this exception